keras_cv/applications/object_detection/retina_net/retina_net_test.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+from absl.testing import parameterized
+
+import keras_cv
+
+
+class RetinaNetTest(tf.test.TestCase, parameterized.TestCase):
```

Comment:
Some other potential cases to consider:

- include_rescaling is successfully passed to the backbone
- Custom label encoders, feature pyramids, prediction decoders work (at least for construction)
- model.compile succeeds for tests where the model is initialized without error

Reply:
> Custom label encoders, feature pyramids, prediction decoders work (at least for construction)

lets leave this as a TODO.  In a follow up PR I'll be wrapping these layers into library layers that are re-usable, so when we do that we can make this a real API.

Reply:
Added a comment about include_rescaling, it wont work until I move to keras_cv.models